simularium/simularium-viewer
# Binding simulation

Problem
=======
I built a prototype for the education module. First here, and then in a new repo. In order to install the viewer and make use of the front end simulation interface, I had to make a number of changes. 
#### Issues
1. the other repo isn't using babel
2. If you remount the viewer component, it parents a second canvas (only known because of `React.StrictMode`)
3. A number of types/interfaces were needed by the installing app but weren't exposed 
4. The other app couldn't resolve the urls for the web workers  

Solution
========
1. moved some runtime babel deps from dev deps into regular 
2. Added check to see of the canvas had already been added to the parent div
3. Exported interfaces 
4. added `{ type: "module" }` for the URL declarations 

## Type of change
Please delete options that are not relevant.

* Bug fix (non-breaking change which fixes an issue)

Steps to Verify:
----------------
Since these changes are largely to enable installation by another app, to truly test it you need to also download the [other repo](https://github.com/simularium/binding-sim-edu) and install this branch as the viewer. Short of that, everything else in the test bed app should behave normally.
